client/src/modules/stock/StockForm.service.js

```diff
@@ -125,5 +126,57 @@ function StockFormService(StockItem, Store, AppCache, Session, $timeout) {
     return Object.keys(this.cache).length > 0;
   };
 
+  /**
+   * @method catchDuplicatedLots
+   *
+   * @description
+   * this method catch duplicated row and emit notification on the row
+   */
+  StockForm.prototype.catchDuplicatedLots = function catchDuplicatedLots(_store) {
+    const ROW_ERROR_FLAG = bhConstants.grid;
+    let selectedLots;
+    let doublonDetectedLine;
+
+    if (hasDuplicatedLots(_store)) {
+      // notify on the concerned row
+      errorLineHighlight(doublonDetectedLine, _store);
+      return true;
+    }
+
+    function errorLineHighlight(rowIdx, store) {
+      // set and unset error flag for allowing to highlight again the row
+      // when the user click again on the submit button
+      const row = store.data[rowIdx];
+      row[ROW_ERROR_FLAG] = true;
+      $timeout(() => {
+        row[ROW_ERROR_FLAG] = false;
+      }, 1000);
+    }
+
+    // update the list of selected lots
+    function refreshSelectedLotsList(store) {
+      selectedLots = store.data
+        .filter(item => item.lot && item.lot.uuid)
+        .map(item => item.lot.uuid);
+    }
+
+    // detect the presence of duplicated lots
+    function hasDuplicatedLots(store) {
+      refreshSelectedLotsList(store);
```

Comment:
Instead of making a variable `selectedLots` in the upper scope, let's just make it a `const` in this scope   Like this:
```js

    function refreshSelectedLotsList(store) {
      return store.data
        .filter(item => item.lot && item.lot.uuid)
        .map(item => item.lot.uuid);
    }

 function hasDuplicatedLots(store) {
      const selectedLots = refreshSelectedLotsList(store);
      // etc...
 }
```